File: exclusaodedadosJAVA/eventoseparticipantes.java
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.PreparedStatement;
import java.sql.SQLException;
//Nome:Joao Carlos

public class eventoseparticipantes {
    public static void main(String[] args) {
        final String url = "jdbc:mysql://localhost:3306/nomedobanco";
        final String user = "nome de usuario";
        final String password = "senha";

        try (Connection conn = DriverManager.getConnection(url, user, password)) {
            conn.setAutoCommit(false);

            int idEvento = 1; // Id do evento a ser excluído

            // Remove todas as associações do evento com participantes
            String sql = "DELETE FROM participantes WHERE evento_id = ?";
            try (PreparedStatement pstmt = conn.prepareStatement(sql)) {
                pstmt.setInt(1, idEvento);
                pstmt.executeUpdate();
            }

            // Exclui o evento
            sql = "DELETE FROM eventos WHERE id = ?";
            try (PreparedStatement pstmt = conn.prepareStatement(sql)) {
                pstmt.setInt(1, idEvento);
                pstmt.executeUpdate();
            }

            conn.commit();
            System.out.println("Evento e participantes associados excluídos com sucesso!");

        } catch (SQLException e) {
            System.out.println("Erro ao excluir evento e participantes associados: " + e.getMessage());
        }
    }
}

File: exclusapdedadosPHP/eventoseparticipantes.php
<?php // nome: maxwell
$servername = "localhost";
$username = "USER";
$password = "PASS";
$dbname = "DATABASE";

$conn = new mysqli($servername, $username, $password, $dbname);

if ($conn->connect_error) {
    die("Erro de conexão: " . $conn->connect_error);
}

$id_evento = 1; // ID do evento que você deseja excluir

try {
    $conn->begin_transaction();

    // Exclui os participantes do evento específico na tabela `participantes`
    $sql = "DELETE FROM participantes WHERE id_evento = ?";
    $stmt = $conn->prepare($sql);
    $stmt->bind_param("i", $id_evento);
    $stmt->execute();

    // Exclui o evento específico da tabela `eventos`
    $sql = "DELETE FROM eventos WHERE id = ?";
    $stmt = $conn->prepare($sql);
    $stmt->bind_param("i", $id_evento);
    $stmt->execute();

    $conn->commit();

    echo "Evento e participantes excluídos com sucesso!";

} catch (Exception $e) {
    echo "Erro: " . $e->getMessage();

} finally {
    $stmt->close();
    $conn->close();
}
?>